graphula-core/src/Graphula.hs

```diff
@@ -430,34 +410,38 @@ nodeEditWith dependencies edits =
 
   > nodeEdit @Dog (ownerId, veterinarianId)
 -}
-nodeWith :: forall a m. GraphulaContext m '[a] => Dependencies a -> m (Node m a)
+nodeWith :: forall a m. GraphulaContext m '[a] => Dependencies a -> m (Entity a)
 nodeWith = flip nodeEditWith id
 
 {-|
   Generate and edit a value that does not have any dependencies.
 
   > nodeEdit @Dog $ \dog -> dog {name = "fido"}
 -}
-nodeEdit :: forall a m. (GraphulaContext m '[a], Dependencies a ~ ()) => (a -> a) -> m (Node m a)
+nodeEdit :: forall a m. (GraphulaContext m '[a], Dependencies a ~ ()) => (a -> a) -> m (Entity a)
 nodeEdit = nodeEditWith ()
 
 -- | Generate a value that does not have any dependencies
 --
 -- > node @Dog
-node :: forall a m. (GraphulaContext m '[a], Dependencies a ~ ()) => m (Node m a)
+node :: forall a m. (GraphulaContext m '[a], Dependencies a ~ ()) => m (Entity a)
 node = nodeWith ()
 
 attemptsToInsertWith
   :: forall a m . (Typeable a, GraphulaContext m '[a])
   => Int
   -> m a
-  -> m (Node m a)
+  -> m (Entity a)
 attemptsToInsertWith attempts source
   | 0 >= attempts =
       throwIO . GenerationFailureMaxAttempts $ typeRep (Proxy :: Proxy a)
   | otherwise = do
     value <- source
-    insert value >>= \case
+    mKey <- liftIO $ generate genEntityKey
+    mNode <- runDB $ getEntity =<< case mKey of
+      Nothing -> insert value
```

Comment:
Recall that we need to check unique constraints here. `insert` will throw if unique constraints are violated.